### PR TITLE
Specter2: Webpack upgrade

### PR DESCRIPTION
**Pull request by lynnerang**
Thanks again for forging the way on this, Grant!

### CODE REVIEW

File: specter2/package.json [hunk 1]
```diff
@@ -37,37 +37,39 @@
...
+    "eslint-plugin-jsdoc": "^39.3.6",
+    "eslint-plugin-svelte3": "^4.0.0",
+    "html-inline-script-webpack-plugin": "^3.1.0",
+    "html-loader": "^4.1.0",
+    "html-webpack-plugin": "^5.5.0",
     "lint-staged": "^11.0.0",
-    "sass": "^1.37.5",
-    "sass-loader": "^10.1.1",
-    "style-loader": "^2.0.0",
+    "pre-commit": "^1.2.2",
```
Comment (gblakeman) [inline specter2/package.json:58]: You can kill `pre-commit` for your mono-repo setup (it only works on standard repos where the project is at the root).

File: specter2/src/Crawler.ts [hunk 2]
```diff
@@ -8,13 +8,13 @@ const {
...
 /**
- * @description A class to handle traversing an array of selected items and return useful items
+ * A class to handle traversing an array of selected items and return useful items
  * (child nodes, first in selection, node position, spacing between/across nodes).
  *
  * @class
  * @name Crawler
  *
- * @constructor
+ * @class
```
Comment (gblakeman) [inline specter2/src/Crawler.ts:17]: I think this second `@class` is redundant (already on `:14`).

File: specter2/tsconfig.json [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
+    "noImplicitAny": false, // todo: enable when there's time to address new issues
     "types": [
       "node",
-      "@pyoner/svelte-types",
+      "svelte",
       "@figma/plugin-typings"
     ]
   }
-}
+}
```
Comment (gblakeman) [inline specter2/tsconfig.json:17]: GitHub wants an empty line at the end.

File: specter2/webpack.config.js [hunk 4]
```diff
@@ -13,63 +13,42 @@ module.exports = (env, argv) => [{
...
-      { test: /\.tsx?$/, use: 'ts-loader', exclude: /node_modules/ },
+      {
+        test: /\.tsx?$/,
+        use: 'ts-loader',
+        exclude: /node_modules/,
+      },
+
+      // Force data update module through Typescript loader
+      // --- default webpack config does not like js modules
+      {
```
Comment (gblakeman) [inline specter2/webpack.config.js:47]: This block was only needed for my `run-data-update.mjs` module. I think you can safely remove it. I’ll update the comment in Realish to more explicitly call out that it’s a requirement for that specific file.

File: specter2/.eslintrc.js [hunk 5]
```diff
@@ -106,4 +112,4 @@ module.exports = {
       }
     }
   ]
-}
+}
```
Comment (gblakeman) [inline specter2/.eslintrc.js:115]: GitHub wants an empty line at the end.

File: specter2/package.json [hunk 6]
```diff
@@ -37,37 +37,39 @@
...
+    "svelte-loader": "^3.1.3",
+    "svelte-preprocess": "^4.10.7",
+    "ts-loader": "^9.3.1",
+    "typescript": "^4.7.4",
+    "url-loader": "^4.1.1",
+    "webpack": "^5.73.0",
+    "webpack-cli": "^4.10.0"
   },
   "dependencies": {
     "caniuse-lite": "^1.0.30001208",
```
Comment (gblakeman) [inline specter2/package.json:75]: Is this dep still used?

I also noticed `uuid-random` is still installed. I added that a couple years ago (possibly by mistake 🤦🏻‍♂️). I don't think it’s getting used for anything.

**Review by gblakeman — APPROVED** [review]
🌈🦄

File: specter2/webpack.config.js (hunk 4 above)
Comment (lynnerang) [reply specter2/webpack.config.js:47]: cool, thanks for the heads-up!

File: specter2/package.json (hunk 6 above)
Comment (lynnerang) [reply specter2/package.json:75]: nope doesn't look like it for either, just removed

**Pull request merged**